File: iNTRODUCCION-POO/producto.py
class Producto:
    def __init__(self, nombre, precio, stock):
        self.nombre = nombre
        self.precio = precio
        self.stock = stock

    def vender(self, cantidad):
        if cantidad <= 0:
            print("La cantidad a vender debe ser mayor que cero.")
            return
        if cantidad > self.stock:
            print("No hay suficiente stock para vender esa cantidad.")
        else:
            self.stock -= cantidad
            print(f"Se vendieron {cantidad} unidades de {self.nombre}.")
    def reabastecer(self, cantidad):
        if cantidad <= 0:
            print("La cantidad a reabastecer debe ser mayor que cero.")
            return
        self.stock += cantidad
        print(f"Se reabastecieron {cantidad} unidades de {self.nombre}.")

File: iNTRODUCCION-POO/producto.java
public class Producto {
    private String nombre;
    private double precio;
    private int stock;

    public Producto(String nombre, double precio, int stock) {
        this.nombre = nombre;
        this.precio = precio;
        this.stock = stock;
    }

    public void vender(int cantidad) {
        if (cantidad <= 0) {
            System.out.println("La cantidad a vender debe ser mayor que cero.");
            return;
        }
        if (cantidad > stock) {
            System.out.println("No hay suficiente stock para vender esa cantidad.");
        } else {
            stock -= cantidad;
            System.out.println("Se vendieron " + cantidad + " unidades de " + nombre + ".");
        }
    }

    public void reabastecer(int cantidad) {
        if (cantidad <= 0) {
            System.out.println("La cantidad a reabastecer debe ser mayor que cero.");
            return;
        }
        stock += cantidad;
        System.out.println("Se reabastecieron " + cantidad + " unidades de " + nombre + ".");
    }

    public String getNombre() {
        return nombre;
    }

    public double getPrecio() {
        return precio;
    }

    public int getStock() {
        return stock;
    }
}